mongoose.c:2762:37: error: format specifies type 'unsigned short' but the argument has type 'int' [-Werror,-Wformat]
       inet_ntoa(sa->sin.sin_addr), ntohs(sa->sin.sin_port)));

We could either cast ntohs to an (unsigned short) or we can remove the -Werror
I elected to do the latter
kencu@macports.org


--- webserver/Makefile.orig	2019-02-25 20:40:46.000000000 -0800
+++ webserver/Makefile	2019-02-25 20:40:56.000000000 -0800
@@ -2,7 +2,7 @@
 MODULE_CFLAGS=-DMG_DISABLE_DAV_AUTH -DMG_ENABLE_HTTP_CGI=0 -DMG_ENABLE_THREADS=0 -DMG_ENABLE_HTTP_SSI=0
 
 SOURCES = main.c mongoose.c
-CFLAGS = -O2 -W -Wall -Werror -Wno-unused-function $(CFLAGS_EXTRA) $(MODULE_CFLAGS)
+CFLAGS = -O2 -W -Wall -Wno-unused-function $(CFLAGS_EXTRA) $(MODULE_CFLAGS)
 
 ifeq ($(OS), Windows_NT)
 CFLAGS += -lws2_32 -Wno-format
